packages/payment-processor/test/payment/erc-20-private-payment-hinkal.test.ts

```diff
@@ -0,0 +1,256 @@
+import { BigNumber, ethers, providers, Wallet } from 'ethers';
+import { RequestNetwork, Types, Utils } from '@requestnetwork/request-client.js';
+import { CurrencyTypes } from '@requestnetwork/types';
+import { EthereumPrivateKeySignatureProvider } from '@requestnetwork/epk-signature';
+import {
+  addToHinkalStore,
+  hinkalStore,
+  payErc20FeeProxyRequestFromHinkalShieldedAddress,
+  payErc20ProxyRequestFromHinkalShieldedAddress,
+  sendToHinkalShieldedAddressFromPublic,
+} from '../../src/payment/erc-20-private-payment-hinkal';
+import { getErc20Balance } from '../../src/payment/erc20';
+
+// Constants to configure the tests
+const currentNetwork: CurrencyTypes.ChainName = 'base';
+const currencyAddress = '0x833589fCD6eDb6E08f4c7C32D4f71b54bdA02913'; // USDC
+const currentCurrenyType = Types.RequestLogic.CURRENCY.ERC20;
+const currencyAmount = ethers.utils.parseUnits('0.000001', 6).toBigInt();
+const currentGateway = 'https://sepolia.gateway.request.network';
+const fee = ethers.utils.parseUnits('0.000001', 6).toString();
+
+// Payer:
+// 1) sends funds from his shielded address to EOA of the payee
+// 2) sends funds from his EOA to the shielded address of the payee
+// 3) sends funds from hiw EOA to his own shielded address
+// Note: to successfully run the tests, you will need to top up payer's EOA address with Eth on base
+const payerMnemonic = 'sand win seed crucial film antique adapt start pupil gallery edge collect';
+const payerAddress = '0x01aC930fAb20482E5b7eAbF892DF57141c29de5F';
+// Payee:
+// 1) receives funds on her public EOA address
+// 2) receives funds on her shielded address
+// The private key of a public address grant ownership of the corresponding shielded address. In @hinkal/common, a single public address can have only one shielded address.
+const payeeAddress = '0x44DC1e666C1ca6717849efE19eAC72AD83cFf5d2';
+const payeePrivateKey = '0x3abdcb3d6d6c302a7943715d0b975ae1377d7d1d188820f6cd57b6f13fb5b0e0';
+const payeeShieldedAddress =
+  '142590100039484718476239190022599206250779986428210948946438848754146776167,0x096d6d5d8b2292aa52e57123a58fc4d5f3d66171acd895f22ce1a5b16ac51b9e,0xc025ccc6ef46399da52763a866a3a10d2eade509af27eb8411c5d251eb8cd34d';
+
+const RPC_URL = 'https://mainnet.base.org'; // Blockchain RPC endpoint for the Base network
+
+jest.setTimeout(1000000); // Set Jest timeout for asynchronous operations (e.g., blockchain calls)
```

Comment:
This sets the default timeout for all tests in this file, see https://jestjs.io/docs/jest-object#jestsettimeouttimeout
1000000ms = 1000s = 16m per test, that seems a bit much
I'd advise reducing it to 30s or 60s max per test, so 30000 or 60000
```suggestion
jest.setTimeout(30000); // Set the default timeout interval (in milliseconds) for all tests and before/after hooks in this test file.
```

Reply:
I tried 30000 but all four tests failed so I reverted back to 16m